jupyterhubutils/lsstmgr/authmanager.py

```diff
@@ -0,0 +1,95 @@
+'''Class to hold LSST-specific authentication/authorization details
+and methods.
+
+Most of this was formerly held in the JupyterHub config as classes defined
+in '10-authenticator.py'.
+'''
+import random
+
+from ..utils import make_logger, list_duplicates
+
+
+class LSSTAuthManager(object):
+    authenticator = None
+    group_map = {}  # key is group name, value is group id
+
+    def __init__(self, *args, **kwargs):
+        self.log = make_logger()
+        self.log.debug("Creating LSSTAuthManager.")
+        self.parent = kwargs.pop('parent')
+        self.authenticator = self.parent.authenticator
+
+    def _group_merge(self, groups_1, groups_2):
+        # First just merge them
+        grps = []
+        grps.extend(groups_1)
+        grps.extend(groups_2)
+        # Naive deduplication
+        grps = list(set(grps))
+        grpnames = [x.split(':', 1)[0] for x in grps]
+        gnset = set(grpnames)
+        # Check for need to do less-naive dedupe
+        if len(gnset) != len(grpnames):
+            # We have a collision
+            grps = self._deduplicate_groups(grps, grpnames)
+        return grps
+
+    def _deduplicate_groups(self, grps):
+        grpsplits = [x.split(':', 1) for x in grps]
+        grpnames = [x[0] for x in grpsplits]
+        grpnums = [x[1] for x in grpsplits]
+        flist = list_duplicates(grpnames)
+        for name, positions in flist:
+            i = 1
+            for p in positions[1:]:
+                # start with group_2 (leave the first one alone)
+                i += 1
+                grps[p] = grpnames[p] + "_" + str(i) + ":" + grpnums[p]
+        return grps
+
+    def check_membership(self):
+        allowed_groups = self.authenticator.allowed_groups.split(",")
```

Comment:
put allowed and forbidden groups on config object and maybe set them on lsst_configure once instead of authenticator class doing it each time